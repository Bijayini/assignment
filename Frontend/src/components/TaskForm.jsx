import React, {Fragment} from "react";

import labels from '../labels/Task.labels';
import FormField from "../ui-components/FormField";

const TaskForm  = ({title, description, status, handleChange, hasError})=>{
    return(
        <Fragment>
            <FormField label={labels.title} labelFor={title}>
                <input type="text" id="title" name="title" value={title} onChange={handleChange} />
            </FormField>
            <FormField label={labels.description} labelFor={description}>
                <textarea id="description" name="description" value={description} onChange={handleChange} />
            </FormField>
            <FormField label={labels.status} labelFor={status}>
                <select value={status} name="status" onChange={handleChange}>
                    <option value="">{labels.select_status}</option>
                    <option value="in progress">{labels.in_progress}</option>
                    <option value="completed">{labels.completed}</option>
                    <option value="urgent">{labels.urgent}</option>
                </select>
            </FormField>
            {hasError &&<p className="error-message">{labels.please_fill_all_required_field}</p>}
        </Fragment>
    );
};

export default TaskForm;
