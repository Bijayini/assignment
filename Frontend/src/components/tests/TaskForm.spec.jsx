import React from 'react';
import { shallow } from 'enzyme';

import TaskForm from '../TaskForm';

describe('<TaskForm />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      description: 'some desc',
      status: 'some value',
      title: 'some title',
      handleChange: jest.fn(),
      hasError:true
    };
    component = shallow(<TaskForm {...props}> Hello </TaskForm>);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});