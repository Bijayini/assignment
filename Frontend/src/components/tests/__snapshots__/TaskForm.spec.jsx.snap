// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskForm /> should match with snapshot 1`] = `
"<Fragment>
  <FormField label=\\"Title\\" labelFor=\\"some title\\">
    <input type=\\"text\\" id=\\"title\\" name=\\"title\\" value=\\"some title\\" onChange={[Function: mockConstructor]} />
  </FormField>
  <FormField label=\\"Description\\" labelFor=\\"some desc\\">
    <textarea id=\\"description\\" name=\\"description\\" value=\\"some desc\\" onChange={[Function: mockConstructor]} />
  </FormField>
  <FormField label=\\"Status\\" labelFor=\\"some value\\">
    <select value=\\"some value\\" name=\\"status\\" onChange={[Function: mockConstructor]}>
      <option value=\\"\\">
        Select Status
      </option>
      <option value=\\"in progress\\">
        In Progress
      </option>
      <option value=\\"completed\\">
        Completed
      </option>
      <option value=\\"urgent\\">
        Urgent
      </option>
    </select>
  </FormField>
  <p className=\\"error-message\\">
    Please fill all required field
  </p>
</Fragment>"
`;
